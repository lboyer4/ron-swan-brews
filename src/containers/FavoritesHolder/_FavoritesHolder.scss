.no-favorites {
	font-weight: bold;
	display: block;
	width: 100%;
	font-size: 30px;
	color: #BA5624;
	text-shadow: 2px 2px 2px black;
	letter-spacing: 2px;
}

.image-container {
	display: inline-block;
	align-content: center;
	justify-content: center;
}