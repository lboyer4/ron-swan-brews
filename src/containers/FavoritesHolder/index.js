import React from 'react';
import { connect } from 'react-redux';
import Card from '../Card';
import PropTypes from 'prop-types';

export const FavoritesHolder = (props) => {
  const favorite = props.breweries.map(brewery => {
      if(brewery.favorited) {
      return <Card {...brewery} key={brewery.id} />
    };
  });
  return(
    <section className='main'>
      <section className='image-container'>
        {favorite}
      </section>
    </section>
  )
};

export const mapStateToProps = (state) => ({
  breweries: state.breweries
});

export default connect(mapStateToProps)(FavoritesHolder);

FavoritesHolder.propTypes = {

breweries: PropTypes.array,
  dispatch: PropTypes.func,
  history: PropTypes.object,
  location: PropTypes.object,
  match: PropTypes.object
}