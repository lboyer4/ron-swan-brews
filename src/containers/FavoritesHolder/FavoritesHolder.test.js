import React from 'react';
import { shallow } from 'enzyme';
import { FavoritesHolder } from '../FavoritesHolder';

describe('FavoritesHolder'
, () => {
	let wrapper;
	let mockBreweries;

	beforeEach(() => {
		mockBreweries = [
			{brewery: 'one',
			favorited: false,
			id: 1},
			{brewery: 'one',
			favorited: false,
			id: 2}
		];

		wrapper = shallow(
			<FavoritesHolder 
				breweries={mockBreweries}
			/>
		)
	});

	it('should match snapshots', () => {
		expect(wrapper).toMatchSnapshot()
	})
});