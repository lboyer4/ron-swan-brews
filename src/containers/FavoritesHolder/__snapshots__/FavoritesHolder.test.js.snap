// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesHolder should match snapshots 1`] = `
<section
  className="main"
>
  <section
    className="image-container"
  />
</section>
`;
